adding Cmake functionality

andik opened the pull request:
added CMakelists based on https://github.com/arunoda/cmake-boilerplate.

```sh
mkdir _build
cd _build
cmake ..
make
```
will compile the lib and the standalone example, no iOS yet

andik commented:
the exmaples build is currently OSX only. I need to add some parts of RTAudio buildsystem to configure this correctly. the lib itself should build without problems.
But I think this is a good starting point

morganpackard commented:
Awesome!
Does this generate an xcode project, or just a lib?

On Thu, Jan 29, 2015 at 2:23 AM, andik <notifications@github.com> wrote:

> the exmaples build is currently OSX only. I need to add some parts of
> RTAudio buildsystem to configure this correctly. the lib itself should
> build without problems.
> But I think this is a good starting point
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/TonicAudio/Tonic/pull/255#issuecomment-71981313>.
>



-- 
===============
Morgan Packard
cell: (720) 891-0122
twitter: @morganpackard

morganpackard commented:
@andik I have no doubt that this is a great enhancement to the workflow. However, relying on cmake rather than XCode or Visual Studio for compilation is a pretty major change in workflow for me and everyone else using Tonic. If this is intended to be a replacement for compiling with XCode/Visual studio, we'll need to consider this PR very carefully.

LB-- commented:
Huh? Doesn't CMake generate Visual Studio projects on Windows by default?

morganpackard commented:
@LB-- That's sort of what I thought, based on my extreme superficial understanding of CMake, but it sounds like what @andik has created is something that compiles outside of Visual Studio.

LB-- commented:
In order for CMake to succeed I had to change this line:
```cmake
		string(REGEX REPLACE ".*\\" "" testname "${filename}" )
```
To this:
```cmake
		string(REGEX REPLACE ".*\\\\" "" testname "${filename}" )
```
After that,
```text
C:\Users\Nicholas\Desktop\Tonic-development\build>cmake ..
-- Building for: Visual Studio 12 2013
-- The C compiler identification is MSVC 18.0.30723.0
-- The CXX compiler identification is MSVC 18.0.30723.0
-- Check for working C compiler using: Visual Studio 12 2013
-- Check for working C compiler using: Visual Studio 12 2013 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler using: Visual Studio 12 2013
-- Check for working CXX compiler using: Visual Studio 12 2013 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Warning (dev) at Tests/CMakeLists.txt:31 (add_executable):
  Policy CMP0037 is not set: Target names should not be reserved and should
  match a validity pattern.  Run "cmake --help-policy CMP0037" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The target name "C:/Users/Nicholas/Desktop/Tonic-development/Tests/main" is
  reserved or not valid for certain CMake features, such as generator
  expressions, and may result in undefined behavior.
This warning is for project developers.  Use -Wno-dev to suppress it.

C:/Users/Nicholas/Desktop/Tonic-development/Tests/main C:/Users/Nicholas/Desktop
/Tonic-development/Tests/main.cpp
-- Configuring done
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/main.vcxproj.
tmp
CMake Error: : System Error: Invalid argument
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/main.vcxproj.
filters.tmp
CMake Error: : System Error: Invalid argument
-- Generating done
-- Build files have been written to: C:/Users/Nicholas/Desktop/Tonic-development
/build
```
So there's still some issues with the `CMakeLists.txt` in `/Tests/`, but the project at least generated.

I opened `/build/Tonic.sln` and once the solution loaded, I did a rebuild-all. It succeeded with [402 warnings](https://gist.github.com/LB--/ab75f54a5fd060b3c27d). I tried running `/build/examples/Standalone/Debug/TonicStandaloneDemo.exe` but got this:
```text
RtApiDummy: This class provides no functionality.


RtApi::openStream: output device parameter value is invalid.
```

So, at least it builds.

morganpackard commented:
@LB-- Thanks for taking a look.

andik commented:
added a commit @LB-- Windows should theoretically work now. I also removed that "main" target...
Still the warnings are remaining... I have really no Idea how to fix it as Cmake seems to add a -Wmost to the XCode project and equivalent to VS Project...

andik commented:
ah and for XCode Projects do
```
mkdir _prj
cd _prj
cmake -G Xcode ..
open Tonic.xcodeproj
```

andik commented:
ok, looks like the warnings will stay until -wmost is manually removed from the xcode build settings... this is awkward, but built in cmake XCode Generator...

LB-- commented:
Something is still a little off with the CMake scripts:
```text
C:\Users\Nicholas\Desktop\Tonic-development\build>cmake ..
-- Building for: Visual Studio 12 2013
-- The C compiler identification is MSVC 18.0.31101.0
-- The CXX compiler identification is MSVC 18.0.31101.0
-- Check for working C compiler using: Visual Studio 12 2013
-- Check for working C compiler using: Visual Studio 12 2013 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler using: Visual Studio 12 2013
-- Check for working CXX compiler using: Visual Studio 12 2013 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Warning (dev) at Tests/CMakeLists.txt:31 (add_executable):
  Policy CMP0037 is not set: Target names should not be reserved and should
  match a validity pattern.  Run "cmake --help-policy CMP0037" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The target name "C:/Users/Nicholas/Desktop/Tonic-development/Tests/main" is
  reserved or not valid for certain CMake features, such as generator
  expressions, and may result in undefined behavior.
This warning is for project developers.  Use -Wno-dev to suppress it.

C:/Users/Nicholas/Desktop/Tonic-development/Tests/main C:/Users/Nicholas/Desktop
/Tonic-development/Tests/main.cpp
CMake Warning (dev) at Tests/CMakeLists.txt:31 (add_executable):
  Policy CMP0037 is not set: Target names should not be reserved and should
  match a validity pattern.  Run "cmake --help-policy CMP0037" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The target name
  "C:/Users/Nicholas/Desktop/Tonic-development/Tests/testmain" is reserved or
  not valid for certain CMake features, such as generator expressions, and
  may result in undefined behavior.
This warning is for project developers.  Use -Wno-dev to suppress it.

C:/Users/Nicholas/Desktop/Tonic-development/Tests/testmain C:/Users/Nicholas/Des
ktop/Tonic-development/Tests/testmain.cpp
-- Using Windows WASAPI
-- Configuring done
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/main.vcxproj.
tmp
CMake Error: : System Error: Invalid argument
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/main.vcxproj.
filters.tmp
CMake Error: : System Error: Invalid argument
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/testmain.vcxp
roj.tmp
CMake Error: : System Error: Invalid argument
CMake Error: Cannot open file for write: C:/Users/Nicholas/Desktop/Tonic-develop
ment/build/Tests/C:/Users/Nicholas/Desktop/Tonic-development/Tests/testmain.vcxp
roj.filters.tmp
CMake Error: : System Error: Invalid argument
-- Generating done
-- Build files have been written to: C:/Users/Nicholas/Desktop/Tonic-development
/build
```
It still generated though, so I tried building again.

![](https://i.imgur.com/VVBA0La.png)

...interesting. Probably because of the weird errors during the CMake generator. Something's definitely worse, though, because this time [the build failed](https://gist.github.com/LB--/5efed08c2460002d09dd).

When I have more time later I might try to resolve some issues myself but as I am new to the project I don't know what state everything is in.

andik commented:
It doesn't find the topic lib, But Xcode does. @LB-- Could you please do a 
    dir /s/b
In your project root and send me that? I have windows only at work and need to know where it put the lib.

andik commented:
Tested in my lunchbreak at work... my installed old Visual Studio doesn't compile Tonic at all... so I cannot verify anything. But I did not have there Write errors you had. May be some IO Issue not related to CMake? Maybe you should remove you whole build folder and try again? I changed the example from including the object files to including the Tonic Lib. And please use a Subfolder...

sorry to say, that I will not be available over the weekend. My girfriend visits me (mostly every weekend) and I'd rather look at her pretty face than my screen ^^

LB-- commented:
Maybe a six-minute video of what I am doing would help?
https://drive.google.com/file/d/0B6gYqBQuTG-9Q29PSi03bHp5Y3M/view?usp=sharing

andik commented:
Thank yo so much for pointing this out. Interesting way to share what you're dong.
I conclude: the targets builds are independent. The library has an error so the example can't be built. I oversaw the error in your old build logs. Except of the "unknown pragma" warnings all warnings are correct and hopefully will be fixed someday.

Could you please change that to
```c++
 SampleTable loadAudioFile(string path, int numChannels){
Tonic::error("loadAudioFile is currently only implemented for Apple platforms.", true);
return NULL;
}
```

and try to build again?

LB-- commented:
Not sure how that's supposed to change anything? The error is on the `return NULL` because two constructors of `SampleTable` can be used and the compiler can't decide which to use.
![image](https://cloud.githubusercontent.com/assets/1296838/5988721/848b67ae-a931-11e4-84a4-c78179015df3.png)
I changed it to `return nullptr;` instead and it compiled successfully. The example still has the same output as before.

andik commented:
Sorry for misleading you. I just didn't think enough before writing ...

Could you please look in the solition if __WINDOWS_WASAPI__ is defined? It defines the driver which rtaudio us using. Your log shows that rtaudio is used but somehow it's not.

andik commented:
I mean if it is defined for the standalone project

LB-- commented:
Yes:
![image](https://cloud.githubusercontent.com/assets/1296838/5992952/cdb1e042-aa04-11e4-8791-6aadf1aa5513.png)

andik commented:
okay, while this is at it should be, it's not getting easier ^^.
From looking at @Dewb 's VS Solution: Could you (please) try the following?

           cmake -DAUDIO_WINDOWS_DS=ON ..

thanks you for helping me out....

In the meantime I have updated the cmakelists a little and most important, found a way to diable the xcode warning mess. (some useless warnings still remaining, but i'll fix that later...)

LB-- commented:
I didn't try https://github.com/andik/Tonic/commit/fb703e7f380f22900233eb9fb059a1934710317b because it didn't look like ti would change anything of importance (I can if you want me to, but I would need to re-patch that `return NULL;` -> `return nullptr;`). Using the `-DAUDIO_WINDOWS_DS=ON` flag gives me some link errors.
![image](https://cloud.githubusercontent.com/assets/1296838/5994906/89920388-aa4a-11e4-8c51-bf9cf34bc5b1.png)
Full error log [here](https://gist.github.com/LB--/c9b1286c6d8cadf21133).

andik commented:
yep, I did forgot to used the RTAudio Liblist, sorry. If you're still willing to test this out, you should download the complete package. the patch you did is included. It did compile under mingw, but without the standalone demo as I have no directX installed. other methods are currently not working. (the RTAudio libs in tonic are older than the cmake file I used and do not support WSAPI, thats why the example was not working...)

LB-- commented:
`cmake ..` runs with no issues at all :+1: no more weird broken projects in the VS solution either.

Everything compiles successfully (still 227 warnings) and the example actually runs and plays an audible sound! It looks like it's all working now.

andik commented:
yeah! Thank you so much for your help! No I can go forward porting the unit tests.

morganpackard commented:
You guys are amazing.

On Mon, Feb 2, 2015 at 9:45 AM, andik <notifications@github.com> wrote:

> yeah! Thank you so much for your help! No I can go forward porting the
> unit tests.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/TonicAudio/Tonic/pull/255#issuecomment-72468250>.
>



-- 
===============
Morgan Packard
cell: (720) 891-0122
twitter: @morganpackard

andik commented:
@morganpackard 
okay, I fixed the last compiler warnings (except #258). If you like you can merge it now.

morganpackard commented:
I'd like to give this a try which directory do I run the cmake command
from? I see you instructions to run it from _build, but where should that
_build directory be located?

On Mon, Feb 2, 2015 at 2:41 PM, andik <notifications@github.com> wrote:

> @morganpackard <https://github.com/morganpackard>
> okay, I fixed the last compiler warnings (except #258
> <https://github.com/TonicAudio/Tonic/issues/258>). If you like you can
> merge it now.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/TonicAudio/Tonic/pull/255#issuecomment-72523028>.
>



-- 
===============
Morgan Packard
cell: (720) 891-0122
twitter: @morganpackard

LB-- commented:
In the root directory create a folder with any name and just run `cmake ..` from there. Both `_build` and `build` are fine.

morganpackard commented:
I"m seeing references to  LazySampleTable.cpp in the xcode project. I think
we need to get those out of there before I can merge this pull request.

Also, please be patient with me. I'd like to look over all of these changes
and test carefully, which will take some time.

On Mon, Feb 2, 2015 at 3:43 PM, Morgan Packard <morgan@morganpackard.com>
wrote:

> I'd like to give this a try which directory do I run the cmake command
> from? I see you instructions to run it from _build, but where should that
> _build directory be located?
>
> On Mon, Feb 2, 2015 at 2:41 PM, andik <notifications@github.com> wrote:
>
>> @morganpackard <https://github.com/morganpackard>
>> okay, I fixed the last compiler warnings (except #258
>> <https://github.com/TonicAudio/Tonic/issues/258>). If you like you can
>> merge it now.
>>
>> --
>> Reply to this email directly or view it on GitHub
>> <https://github.com/TonicAudio/Tonic/pull/255#issuecomment-72523028>.
>>
>
>
>
> --
> ===============
> Morgan Packard
> cell: (720) 891-0122
> twitter: @morganpackard
>



-- 
===============
Morgan Packard
cell: (720) 891-0122
twitter: @morganpackard

morganpackard commented on src/Tonic/Noise.cpp:
```diff
@@ -14,9 +14,9 @@
   namespace Tonic_ {
   
-    PinkNoise_::PinkNoise_() : pinkCount_(1), pinkAccum_(0) {
+    PinkNoise_::PinkNoise_() :pinkAccum_(0),  pinkCount_(1) {
```
@andik Would you mind explaining the purpose of this change and the other similar changes, where the initilizer lists are reordered?

LB-- replied:
Some compilers issue warnings when the initalizer lists are not in the same order as the class members. This is because in C++ members are always initialized in the order they are declared, regardless of the order they appear in initializer lists.

morganpackard commented on lib/TonicLib.xcodeproj/project.pbxproj:
```diff
@@ -699,6 +707,7 @@
 				0183D0671735D0E6004638EB /* BufferFiller.cpp in Sources */,
 				0183D06A1735D0E6004638EB /* CompressorLimiter.cpp in Sources */,
 				0183D06D1735D0E6004638EB /* ControlArithmetic.cpp in Sources */,
+				E1668A7B1A705FCE0008C265 /* LazySampleTable.cpp in Sources */,
```
This definitely needs to be removed from this PR.

morganpackard commented on src/Tonic/SampleTable.h:
```diff
@@ -98,6 +114,16 @@ namespace Tonic {
     void resample(unsigned int frames, unsigned int channels){
       obj->resample(frames, channels);
     }
+      
```
@andik  This work on SampleTable is unrelated to the cmake functionality, and probably doesn't belong in a pull request titled "Adding CMake functionality".

morganpackard commented on examples/Standalone/TonicStandaloneDemo/main.cpp:
```diff
@@ -41,7 +41,7 @@ int main(int argc, const char * argv[])
     
     // --------- MAKE A SYNTH HERE -----------
         
-    ControlMetro metro = ControlMetro().bpm(100);
```
@andik Was 100 bpm too fast for you?

andik commented:
Arg, I always Act before i think... Sorry I'll fix that.

andik commented:
really, 100bpm is quite fast, how should your heart handle such speeds...

Well, I'm really sorry, I wrote a sampletable which can be loaded from a file before switching to the cmake case. I didn't use a clean branch. the issues in standalone main, xcode proj and the sampletable are from that. I'll try to use old versions for them.
All the other stuff, like adding newline, removing semicolons and editing initialiser lists where made to save you the heart attack when compiling and gettting 200 warnings ^^ I thougt this would have a negative influence upon the acceptance of my PR ^^

andik commented on src/Tonic/AudioFileUtils.cpp:
```diff
@@ -91,7 +91,7 @@ namespace Tonic {
   SampleTable loadAudioFile(string path, int numChannels){
     Tonic::error("loadAudioFile is currently only implemented for Apple platforms.");
-    return NULL;
+    return SampleTable();
```
NULL is defined ((void*) 0) in Windows. This is of course a compiler error. If we return an empty buffer instead, we also avoid the crashing of the app which calls loadAudioFile().

andik commented:
@morganpackard 
okay, looks cleaner now

ndonald2 commented:
@andik Thank you SO much for putting this together. I don't ever work in Windows so it's difficult for me to verify anything that affects Windows builds, but I trust your and @morganpackard's judgement on the validity of everything in here. FWIW code changes made to reduce warnings are *always* welcome in my book.

morganpackard commented on src/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+#include directories helps to include header files
```
I'm not crazy about adding a non-source file to a directory called "src". Is there any way this file can be moved, so that we can keep the `src` directory pristine?

morganpackard commented:
Hey @andik. Thanks for answering my questions about your edits, and resolving that xcode project issue. Now, on to my next annoying question:  

How do you propose we as library developers, and everyone who is an end user of Tonic actually use this cmake setup? The value I see in using CMake is that we wouldn't have to manually maintain three separate IDE projects evey time we add a file. The system you've set up doesn't seem to be designed to actually replace the existing xcode and visual studio project. 

If your main motivation for getting cmake running is so that you can create cross-platform, that's great, and I fully support that direction Basically, I support anything you choose for setting up cross-platform tests. I'll just be really grateful to have them in any form.

If the idea with cmake is that every time someone adds a new file to the project, it gets easily incorporated into the projects for all the other platforms, I'm not sure we have a useable system in place yet.

Another thing -- is there a way we can avoid adding the cmake file to the `src` directory? I think, for someone coming to Tonic for the first time, it would be much nicer not to have potentially mysterious files in that directory.

LB-- commented:
I know I'm not the target of your questions, but I'd like to throw in my thoughts too.

What I see with most projects is that the repo only has the CMake files and no IDE-specific project files. People who work on the library keep their own customized IDE projects local to their hard drives. This way everyone gets their way and anyone can build the project.

As for the location and number of `CMakeLists.txt` files, this varies with projects and usually depends on personal preference. I like to have my `CMakeLists.txt` files match my directory structure, so there is one in every folder in the project. Other people do it differently. It depends also somewhat on the way your directory tree is already set up.

For easily adding and removing files, you can have CMake just grab all files that match a pattern, but this is generally discouraged by the CMake community because CMake will not detect that you have added or removed files until you clear the cache and run CMake again. The way CMake knows to run itself again is by checking for changes on the `CMakeLists.txt` files, and so you are intended to add/remove entries as you add/remove files. If you don't care about having to completely re-run CMake each time you add or remove a file, you can have it so that you don't have to edit any `CMakeLists.txt` files when a new file is added or removed.

This is an example project where addition and removal of source files does not require editing `CMakeLists.txt`:

https://github.com/cpluspluscom/ChessPlusPlus/blob/master/CMakeLists.txt#L25-L27

Note however that, addition and removal of files requires re-running the CMake command. For small projects this is fine but it can be annoying if it takes a long time to run and you are frequently adding/removing files.

morganpackard commented:
Thanks @LB--. It looks to me like @andik set the cmake stuff up so that it automatically adds files, which I like. Tonic is pretty small. I don't see running cmake when adding a file as a big deal. 

I don't want to require end users to use cmake. I definitely want to keep the individual IDE projects in the repository, so people can just fire up xcode or visual studio and be up and running quickly. I'd definitely be interested in trying an approach which requires Tonic _developers_ to use cmake (when adding or removing or moving files), but which doesn't require Tonic _users_ to use it.

andik commented:
Hey Morgan, cmake is for the end user. When a C++ Dev comes along "oh this is a nice library" "oh, it builds with cmake" "oh, it works" "oh, I use it for my project". 
Cmake is getting more and more the de-facto standard for C++ Projects including such things like VST, LV Plugins etc. This is why a project really should support it. cmake also handles installation of a library, which I still need to add.
Only that these cmake files are a little complicated, because I did in fact copy, paste, adapt. And I don't know very much of cmake yet.

Beside that, the cmake generated xcode project looks quite good, and I use it for working.
![pastedgraphic-1](https://cloud.githubusercontent.com/assets/147864/6088428/9513f8a0-ae56-11e4-8bcf-9758984c01a0.png) What's not possible seems to be IOS currently. 

The Cmakelist file in `src` could be moved to `lib`

Dewb commented:
Adding my two cents: CMake can be great for the end-user, especially when a library is large and complicated (e.g. Boost.) I think there's less value for end-users for a small library like Tonic that's intended to be incorporated as source library.

I think focusing the CMake implementation initially around the developer workflow, and replacing the existing platform-specific solution files, would be great. I know I've been holding back on merging the polysynth example into master because of fuzziness around how we want to organize the cross-platform project files for example projects. Using the CMake shift as an opportunity to get a consistent plan together would be awesome.

andik commented:
I personally see cmake as an easy way to start with Tonic. One runs cmake and immediately is able to test the example out. Then he can copy the library into his project. I thought about it a little like openframeworks because that was the way I started to learn Tonic. That's the way I designed the cmake system.

morganpackard commented:
I was able to run the cmake stuff successfully on windows. I just tried building using cmake on my mac and got the following errors:

```[ 76%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/Noise.cpp.o
[ 78%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/RampedValue.cpp.o
[ 80%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/RectWave.cpp.o
[ 81%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/Reverb.cpp.o
[ 83%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/RingBuffer.cpp.o
[ 85%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/SampleTable.cpp.o
[ 86%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/SawtoothWave.cpp.o
[ 88%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/SineWave.cpp.o
[ 90%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/StereoDelay.cpp.o
[ 91%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/Synth.cpp.o
[ 93%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/TableLookupOsc.cpp.o
[ 95%] Building CXX object src/CMakeFiles/Tonic.dir/Tonic/TonicFrames.cpp.o
Linking CXX static library libTonic.a
[ 95%] Built target Tonic
Scanning dependencies of target tonictests
[ 96%] Building CXX object tests/CMakeFiles/tonictests.dir/tonictests.cpp.o
Linking CXX executable tonictests
clang: error: no such file or directory: 'CoreAudio;-framework'
clang: error: no such file or directory: 'Accelerate;-framework'
clang: error: no such file or directory: 'AudioToolbox'
make[2]: *** [tests/tonictests] Error 1
make[1]: *** [tests/CMakeFiles/tonictests.dir/all] Error 2
make: *** [all] Error 2```

I'm on the branch andik/development. Any idea what the issue is?

morganpackard commented:
I'd be very wary of adding any additional complications/steps for a new user to use Tonic (see my error above). But if we can get a cmake workflow that works smoothly for Tonic developers, I'd be in favor of adopting that, and not editing individual platform project files by hand any more. Thanks for the two cents @Dewb. I completely agree.

andik commented:
@morganpackard I completely agree with you. But in fact I guess that a lot of people are quite familar with cmake, but providing them additional Makefiles/Xcode Projects etc. is a good Idea, I think.

would you try again, I gues I know the Issue, but I'm not sure, see the diff to see what may be wrong...

Dewb commented:
@morganpackard looks like something is weird with how the "-framework XXX" statements are being concatenated. Using this approach for OSX framework includes might be a better strategy: http://www.cmake.org/Wiki/CMake:HowToUseExistingOSXFrameworks

morganpackard commented:
@dewb Looks like @andik  [resolved the issue I was having](https://github.com/andik/Tonic/commit/04e0ce3373aee39d0851107904a5deb523ea84af) I suspect he meant "superfluous", rather than "superfluid", but I really like the idea of "superfluid quotes"!.

Dewb commented:
Oops, I didn't see the change. Yeah, that fix makes total sense. I also love the idea of superfluidity in punctuation!

andik commented:
I close this because of #264 will bring this too, and I need my development branch ....
